langchain/chains/llm_bash/base.py

```diff
@@ -51,29 +52,62 @@ def output_keys(self) -> List[str]:
 
     def _call(self, inputs: Dict[str, str]) -> Dict[str, str]:
         llm_executor = LLMChain(prompt=self.prompt, llm=self.llm)
-        bash_executor = BashProcess()
+        bash_executor = BashProcess(persistent=self.persistent)
         self.callback_manager.on_text(inputs[self.input_key], verbose=self.verbose)
 
         t = llm_executor.predict(question=inputs[self.input_key])
         self.callback_manager.on_text(t, color="green", verbose=self.verbose)
 
         t = t.strip()
-        if t.startswith("```bash"):
+        if "```bash" in t:
             # Split the string into a list of substrings
-            command_list = t.split("\n")
-            print(command_list)
+            command_list = self.get_code(t)
+            command_list = [s for s in command_list if s]
+
+            if self.verbose:
+                self.callback_manager.on_text("\nCode: ", verbose=self.verbose)
+                self.callback_manager.on_text(command_list, color="yellow", verbose=self.verbose)
 
-            # Remove the first and last substrings
-            command_list = [s for s in command_list[1:-1]]
             output = bash_executor.run(command_list)
 
             self.callback_manager.on_text("\nAnswer: ", verbose=self.verbose)
             self.callback_manager.on_text(output, color="yellow", verbose=self.verbose)
-
         else:
-            raise ValueError(f"unknown format from LLM: {t}")
+            output = t
+            # raise ValueError(f"unknown format from LLM: {t}")
         return {self.output_key: output}
 
+    def get_code_and_remainder(self, t: str) -> str:
```

Comment:
could we add some simple unit tests for all the get_code_ methods

Reply:
done :)